apps/lib/app_provider.c

```diff
@@ -41,6 +44,43 @@ static int opt_provider_load(const char *provider)
     return 1;
 }
 
+/*
+ * Add this call to the start of any application that support a library context.
+ * NOTE: app_load_default_provider() should also be called after the options
+ * have been processed.
+ */
+OPENSSL_CTX *app_new_libctx(void)
```

Comment:
This function *could* be called once and for all in `apps/opensl.c`'s `main()`...

Reply:
If every app uses a non default libctx then that it true.. otherwise it is not.